testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -787,50 +917,169 @@ impl BitcoinRegtestController {
+
+        let tx_size = serialized_tx.bytes.len() as u64;
+        estimated_fees.register_replacement(tx_size);
+        let mut txid = tx.txid().as_bytes().to_vec();
+        txid.reverse();
+
+        debug!("Transaction relying on UTXOs: {:?}", utxos);
+        let txid = Txid::from_bytes(&txid[..]).unwrap();
+        let mut txids = previous_txids.clone();
+        txids.push(txid.clone());
+        let ongoing_block_commit = OngoingBlockCommit {
+            payload,
+            utxos,
+            fees: estimated_fees,
+            txids,
+        };
 
         info!(
-            "Miner node: submitting leader_block_commit op for {} - {}, waiting for its inclusion in the next Bitcoin block",
-            &payload.block_header_hash,
-            public_key.to_hex()
+            "Miner node: submitting leader_block_commit (txid: {}, rbf: {}, total spent: {}, size: {}, fee_rate: {})",
+            txid.to_hex(),
+            ongoing_block_commit.fees.is_rbf_enabled,
+            ongoing_block_commit.fees.total_spent(),
+            ongoing_block_commit.fees.final_size,
+            fee_rate,
         );
 
+        self.ongoing_block_commit = Some(ongoing_block_commit);
+
+        increment_btc_ops_sent_counter();
+
         Some(tx)
     }
 
+    fn build_leader_block_commit_tx(
+        &mut self,
+        payload: LeaderBlockCommitOp,
+        signer: &mut BurnchainOpSigner,
+        _attempt: u64,
+    ) -> Option<Transaction> {
+        // Are we currently tracking an operation?
+        if self.ongoing_block_commit.is_none() {
+            // Good to go, let's build the transaction and send it.
+            let res = self.send_block_commit_operation(payload, signer, None, None, None, &vec![]);
+            return res;
+        }
+
+        let ongoing_op = self.ongoing_block_commit.take().unwrap();
+
+        let _ = self.sortdb_mut();
+        let burnchain_db = self.burnchain_db.as_ref().expect("BurnchainDB not opened");
+
+        for txid in ongoing_op.txids.iter() {
+            let mined_op = burnchain_db.get_burnchain_op(txid);
+            if mined_op.is_some() {
+                // Good to go, the transaction in progress was mined
+                debug!("Was able to retrieve ongoing TXID - {}", txid);
+                let res =
+                    self.send_block_commit_operation(payload, signer, None, None, None, &vec![]);
+                return res;
+            } else {
+                debug!("Was unable to retrieve ongoing TXID - {}", txid);
+            }
+        }
+
+        // Stop as soon as the fee_rate is 1.25 higher, stop RBF
+        if ongoing_op.fees.fee_rate > (self.config.burnchain.satoshis_per_byte * 150 / 100) {
+            let res = self.send_block_commit_operation(payload, signer, None, None, None, &vec![]);
+            return res;
+        }
+
+        // Did a re-org occurred since we fetched our UTXOs
+        if let Err(e) = burnchain_db.get_burnchain_block(&ongoing_op.utxos.bhh) {
```

Comment:
I don't think this is ever going to fail.  We don't delete data from `BurnchainDB`, so a sibling Bitcoin block's block hash (and all its associated ops) would continue to be accessible through `get_burnchain_block`.

Reply:
I see. I added this condition to have a green light from the test `bitcoind_forking_test`. Do you think we should leave that condition, or rewrite that test?

Reply:
I think the error path here can arise and is definitely worth handling, but I think the check you want here is to verify whether or not `ongoing_op.utxos.bhh` is still on the canonical burnchain fork.  You can tell by either looking at the SPV headers, or by looking at the sortition DB.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2383/commits/b03495b2c652bc1cfb084e089d590fc7e04e1446